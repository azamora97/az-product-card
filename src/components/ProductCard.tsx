import React, { createContext, CSSProperties, JSX } from 'react';
import styles from '../styles/styles.module.css';
import { useProduct } from '../hooks/useProduct';
import { ProductContextProps, Product, onChangeArg, InitialValues, ProductCardHandlers } from '../interface/Interfaces';


export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

export interface Props {
  product: Product;
  children: ( args: ProductCardHandlers) => JSX.Element;
  className?: string;
  style?: CSSProperties;
  onChange?: (args: onChangeArg) => void;
  value?: number;
  initialValues?: InitialValues
}


export const ProductCard = ({ product, children, className, style, onChange, value, initialValues }: Props) => {

  const { counter, increaseBy, maxCount, isMaxCountReached, reset } = useProduct( {onChange, product, value, initialValues} ); 
  
  return (
    <Provider value={{ 
        counter,
        product,
        maxCount,
        increaseBy
    }}>
        <div 
          className={ `${styles.productCard} ${className}` }
          style={ style }
        >
            { children({
                count: counter,
                isMaxCountReached,
                maxCount: initialValues?.maxCount,
                product,
            
                increaseBy,
                reset 
              }) 
            }
        </div>
    </Provider>
  )
}