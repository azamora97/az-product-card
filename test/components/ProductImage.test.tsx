import React from 'react';
import renderer from 'react-test-renderer';
import { product2 } from '../data/products';
import { ProductCard } from '../../src/components/index';
import { ProductImage } from '../../src/components/ProductImage';


// obsoleto, se debe usar jest 
describe("ProductImage", () => {
    
    test('debe de mostrar el componente correctamente con la imagen personalizada', () => {

        const wrapper = renderer.create(
            <ProductImage  img='https://holamundo.jpg' />
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
        
    });

    test("debe de mostrar el componente con la imagen del producto", () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {
                    () => (
                        <ProductImage />
                    )
                }
            </ProductCard>
        )

        expect( wrapper.toJSON() ).toMatchSnapshot();
    });

})