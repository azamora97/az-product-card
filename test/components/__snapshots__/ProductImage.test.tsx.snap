// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage debe de mostrar el componente con la imagen del producto 1`] = `null`;

exports[`ProductImage debe de mostrar el componente correctamente con la imagen 1`] = `null`;

exports[`ProductImage debe de mostrar el componente correctamente con la imagen personalizada 1`] = `null`;

exports[`ProductTitle debe de mostrar el componente con el nombre del producto 1`] = `null`;

exports[`ProductTitle debe de mostrar el componente correctamente con la imagen 1`] = `null`;
